api/models/adoptionVisit.js

```diff
@@ -0,0 +1,56 @@
+const mongoose = require('mongoose');
+const Joi = require('joi');
+
+
+//ADOPTION VISIT SCHEMA 
+
+const adoptionVisitSchema = new mongoose.Schema({
+    _id: mongoose.Types.ObjectId,
+    visitDate: {
+        type: Date,
+        required: true
+    },
+    visitTime: {
+        type: String, // inny sposób??
+        required: true,
+        match: /^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$/ //Time Format HH:MM 24-hour with leading 0
+    },
+    duration: { // 30-120 minut lub enum: [30, 60, 90, 120],
+        type: Number,
+        min: 30,
+        max: 120,
+        required: true
+    },
+    animalID : {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'Animal',
+    },
+    userID: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User', // jak odwołać się do ADMIN - isAdmin: false
+        required: true
+    },
+    employeeID: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User', // jak odwołać się do ADMIN - isAdmin: true
+        required: true
+    }
```

Comment:
Tutaj tak jak rozmawialiśmy wydaje mi się, że najlepiej będzie dodać jako typ wymaganych danych schemat użytkownika-admina. Chyba, że ktoś będzie miał inny, lepszy pomysł.

Reply:
Problem jest taki, że nie mamy użytkownika admina, jest tylko właściwość usera isAdmin. Przegadajmy to na callu, trzeba jakoś sprawdzać czy user o podanym id ma pole isAdmin = true.
Może walidacja w joi tu się przyda?